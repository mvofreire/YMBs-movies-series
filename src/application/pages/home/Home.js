import React, { Fragment } from "react";
import { HomeBanner } from "./components";
import { ListEntertainment } from "components";
import { Divider } from "antd";
import { HomeContext, HomeContextProvider } from "contexts/home";

const Home = ({ data }) => (
  <HomeContextProvider>
    <HomeBanner />
    <Divider />
    <HomeContext.Consumer>
      {props => <ListEntertainment {...props} />}
    </HomeContext.Consumer>
  </HomeContextProvider>
);

export default Home;
