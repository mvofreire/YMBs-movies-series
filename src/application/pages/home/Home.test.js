import React from "react";
import { shape } from "prop-types";
import ReactDOM from "react-dom";
import Home from "./Home";
import { shallow, mount } from "enzyme";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {}
  }
};

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) }
});

function mountWrap(node) {
  return mount(node, createContext());
}

function shallowWrap(node) {
  return shallow(node, createContext());
}

describe("HomePage", () => {
  //   it("Render with no error", () => {
  //     const div = document.createElement("div");
  //     ReactDOM.render(<Home />, div);
  //   });

  it("Wrapper must match snapshot ", () => {
    const wrapper = shallowWrap(<Home />);

    expect(wrapper).toMatchSnapshot();
  });
});
