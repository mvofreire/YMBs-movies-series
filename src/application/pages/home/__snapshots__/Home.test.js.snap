// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Wrapper must match snapshot  1`] = `
<HomeContextProvider>
  <withRouter() />
  <Divider />
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</HomeContextProvider>
`;
